src/validators/user-info-request-validator.ts

```diff
@@ -0,0 +1,68 @@
+import { UserInfoRequestError } from "../errors/user-info-request-error";
+import { IncomingHttpHeaders } from "http";
+import { signedJwtValidator } from "./signed-jwt-validator";
+import { Config } from "../config";
+import { logger } from "../logger";
+
+const AuthorisationHeaderKey: string = "authorization";
+const ValidScopes: string[] = ["openid", "email", "phone"];
+
+export const userInfoRequestValidator = async (
+  userInfoRequestHeaders: IncomingHttpHeaders
+): Promise<
+  | { valid: true }
+  | {
+      valid: false;
+      error: UserInfoRequestError;
+    }
+> => {
+  for (const key of Object.keys(userInfoRequestHeaders)) {
+    userInfoRequestHeaders[key.toLowerCase()] = userInfoRequestHeaders[key];
+  }
+  const accessToken = userInfoRequestHeaders[AuthorisationHeaderKey];
+  let match;
+  try {
+    match = /^Bearer (?<token>.*)$/.exec(accessToken as string);
+  } catch (error) {
+    logger.error("Failed To parse authorisation header.", error);
+    return { valid: false, error: UserInfoRequestError.MISSING_TOKEN };
+  }
+
+  if (!match?.groups?.token) {
+    logger.warn("Missing access token in authorisation header.");
+    return { valid: false, error: UserInfoRequestError.MISSING_TOKEN };
+  }
```

Comment:
have you played around with different cases of this? I feel like this might return invalid token

Reply:
We don't validate jti, iat or sid. I think that's correct for iat and sid. If we wan't to emulate validating the jti field, I guess we'd need to store it when we create a token, because right now we generate a random value.